File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var ComplementDict = new Dictionary<int, int>();

        for(int i = 0; i < nums.Length; i++)
        {
            if(ComplementDict.ContainsKey(nums[i])) {
                return new int[2] {ComplementDict[nums[i]], i};
            }

            ComplementDict[target - nums[i]] = i;
        }

        return null;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]


        l, r = 0, len(nums) - 1

        while l < r:
            mid = (l + r) // 2
            if nums[mid] < nums[r]:
                r = mid
            else:
                l = mid + 1

        return nums[l]


